lxd/instance/drivers/driver_qemu.go

```diff
@@ -2854,6 +2860,14 @@ func (d *qemu) generateQemuConfigFile(mountInfo *storagePools.MountInfo, busName
 				return "", nil, err
 			}
 		}
+
+		// Add Vmgenid device.
+		if len(runConf.VmgenDevice) > 0 {
```

Comment:
I don't think we need the `runConf.VmgenDevice` and I don't like it because the `RunConfig` sections are for passing config between the LXD instance device drivers and the LXD instance drivers, and for VM GenID support, there is no equivalent LXD instance device driver.

Instead here you can just do:

```
if d.architecture == osarch.ARCH_64BIT_INTEL_X86 {
    err = d.addVmgenDeviceConfig(&cfg, runConf.VmgenDevice)
    if err != nil {
       return "", nil, err
    }
}
```

And then update `addVmgenDeviceConfig` to read the VM GenID from the local config, as we were doing before.